### PR TITLE
Add equalize 

### PR DESCRIPTION
**Pull request by abhichou4**
Refers to #1333

### CODE REVIEW

**Comment by Squadrick** [comment]
@abhichou4 Can you fix the merge conflicts?

File: tensorflow_addons/image/color_ops_test.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    assert equalized.dtype == image.dtype
+
+
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+def test_equalize_with_PIL():
+    image = np.random.randint(low=0, high=255, size=(4, 3, 3, 3), dtype=np.uint8)
+    batches = []
+    for i in range(4):
+        batch = ImageOps.equalize(Image.fromarray(image[i, :]))
+        batches.append(batch)
```
Comment (Squadrick) [inline tensorflow_addons/image/color_ops_test.py]: Use list comprehension:
```python
equalized = np.stack([ImageOps.equalize(Image.fromarray(i)) for i in image])
```

File: tensorflow_addons/image/color_ops_test.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
```
Comment (Squadrick) [inline tensorflow_addons/image/color_ops_test.py]: 2019 -> 2020

File: tensorflow_addons/image/color_ops.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+from typing import Optional
+
+
+def equalize_batch(image: TensorLike, data_format: str = "channels_last") -> tf.Tensor:
+    """Implements Equalize function from PIL using TF ops."""
+
+    def scale_channel(image, channel=None):
+        """Scale the data in the channel to implement equalize."""
+        image_dtype = image.dtype
+        image = tf.cast(image[:, :, channel], tf.int32)
```
Comment (Squadrick) [inline tensorflow_addons/image/color_ops.py]: `image[:, :, channel]` assumes that it's `"channels_last"`.

File: tensorflow_addons/image/color_ops_test.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+@pytest.mark.parametrize("shape", [(4, 4,), (4, 4, 1), (4, 4, 3), (5, 4, 4, 3)])
+def test_equalize_dtype_shape(dtype, shape):
+    image = np.ones(shape=shape, dtype=dtype)
+    equalized = color_ops.equalize(tf.constant(image)).numpy()
+    np.testing.assert_equal(equalized, image)
+    assert equalized.dtype == image.dtype
+
+
+@pytest.mark.usefixtures("maybe_run_functions_eagerly")
+def test_equalize_with_PIL():
```
Comment (Squadrick) [inline tensorflow_addons/image/color_ops_test.py]: Add a test for `"channels_first"`.

File: tensorflow_addons/image/color_ops.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            # in the C code for image.point.
+            return tf.clip_by_value(lut, 0, 255)
+
+        # If step is zero, return the original image.  Otherwise, build
+        # lut from the full histogram and step and then index from it.
+        result = tf.cond(
+            tf.equal(step, 0),
+            lambda: image,
+            lambda: tf.gather(build_lut(histo, step), image),
+        )
```
Comment (Squadrick) [inline tensorflow_addons/image/color_ops.py]: You can use a regular python cond. If eager mode, it should work out of the box. 

If graph mode, autograph will do the internal conversion to tf ops.

Comment (Squadrick) [reply tensorflow_addons/image/color_ops.py]: If that fails the test, let me know and then stick to use `tf.cond`.

File: tensorflow_addons/image/color_ops.py [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+import tensorflow as tf
+
+from tensorflow_addons.utils.types import TensorLike
+from tensorflow_addons.image.utils import to_4D_image, from_4D_image
+
+from typing import Optional
+
+
+def equalize_batch(image: TensorLike, data_format: str = "channels_last") -> tf.Tensor:
```
[the remark this answers is not in the document]
Comment (abhichou4) [reply tensorflow_addons/image/color_ops.py]: changed it to `equalize_image`

File: tensorflow_addons/image/color_ops.py (hunk 3 above)
Comment (abhichou4) [reply tensorflow_addons/image/color_ops.py]: Missed that. Adding test-case for `"channel_first"`.

File: tensorflow_addons/image/color_ops.py [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+          shape is not `TensorShape(None)`).
+      data_format: Either 'channel_first' or 'channel_last'
+      name: The name of the op.
+    Returns:
+      Image(s) with the same type and shape as `images`, equalized.
+    """
+    with tf.name_scope(name or "equalize"):
+        image_dims = tf.rank(image)
+        image = to_4D_image(image)
+        image = tf.map_fn(equalize_image, image)
```
Comment (abhichou4) [inline tensorflow_addons/image/color_ops.py]: I missed this. `channel_first` returns the same image in test. Adding a `fn = partial(equalize_image, data_format=data_format)` before calling `map_fn` on it.

**Comment by abhichou4** [comment]
cc @Squadrick

**Pull request merged**